File: taller/IvanRC-taller-Java/src/main/java/factoryPersonasTaller/FactoryPersonas.java
package factoryPersonasTaller;

import modelsTaller.Persona;

public class FactoryPersonas {

    public FactoryPersonas(){}

    private static FactoryPersonas instance = null;

    public static FactoryPersonas getInstance(){
        if(instance == null){
            instance = new FactoryPersonas();
        }
        return instance;
    }

    /**
     * función que sirve para generar una persona, jefe de taller, traajador, chapista o electricista cuyos datos son aleatorios
     * @return el objeto creado con datos aleatorios
     */
    public Persona createRandomPersona(){
        Persona persona = null;

        String[] nombres = {"Iván","Jia","Javier","Ramón","Belén","Cristobal","Rapado","Elkevin","Mayorca","Jorge","Nefer","Sebastian","Álvaro","Romeo"};
        int añosExperiencia = (int)(Math.random()*64);
        int personasACargo = (int)(Math.random()*100);
        int horasDiarias = (int)(Math.random()*9);

        int chance = (int)(Math.random()*100);
        if(chance <= 10){
            persona = new Persona.JefeTaller(nombres[(int)(Math.random()* nombres.length)], añosExperiencia, personasACargo);
        }else{
            if(chance <= 60){
                persona = new Persona.Trabajador(nombres[(int)(Math.random()* nombres.length)], añosExperiencia, horasDiarias);
            }else{
                if(chance <= 80){
                    persona = new Persona.Chapista(nombres[(int)(Math.random()* nombres.length)], añosExperiencia, horasDiarias);
                }else{
                    if(chance <= 90) {
                        persona = new Persona.Electricista(nombres[(int)(Math.random()* nombres.length)], añosExperiencia, horasDiarias);
                    }else{
                        persona = new Persona.NavajaSuiza(nombres[(int)(Math.random()* nombres.length)], añosExperiencia, horasDiarias);
                    }
                }
            }
        }
        return persona;
    }
}

File: ejercicios/IvanRC-Taller-Kotlin/src/main/kotlin/factoryPersonasTaller/FactoryPersonas.kt
package factoryPersonasTaller

import modelsTaller.*

class FactoryPersonas {

    companion object{
        private var instance: FactoryPersonas? = null

        fun getInstance():FactoryPersonas?{
            if(instance == null){
                instance = FactoryPersonas()
            }
            return instance
        }
    }

    /**
     * función que sirve para generar una persona, jefe de taller, traajador, chapista o electricista cuyos datos son aleatorios
     * @return el objeto creado con datos aleatorios
     */
    fun createRandomPersona(): Persona?{
        var persona: Persona? = null

        val nombres = arrayOf("Iván","Jia","Javier","Ramón","Belén","Cristobal","Rapado","Elkevin","Mayorca","Jorge","Nefer","Sebastian","Álvaro","Romeo")
        val añosExepriencia = (1..64).random()
        val personasACargo = (1..100).random()
        val horasDiarias = (1..9).random()

        val chance = (1..100).random()
        if(chance <= 10){
            persona = JefeTaller(nombres.random(), añosExepriencia, personasACargo)
        }else{
            if(chance <= 60){
                persona = Trabajador(nombres.random(), añosExepriencia, horasDiarias)
            }else{
                if(chance <= 90){
                    persona = Chapista(nombres.random(), añosExepriencia, horasDiarias)
                }else{
                    persona = Electricista(nombres.random(), añosExepriencia, horasDiarias)
                }
            }
        }
        return persona
    }
}